docs/wallet/node-api.md

```diff
@@ -0,0 +1,101 @@
+
+```scala mdoc:compile-only
+import akka.actor.ActorSystem
+import org.bitcoins.core.api.NodeApi
+import org.bitcoins.core.crypto.DoubleSha256Digest
+import org.bitcoins.core.protocol.blockchain.Block
+import org.bitcoins.keymanager.bip39.BIP39KeyManager
+import org.bitcoins.node.NodeCallbacks
+import org.bitcoins.node.networking.peer.DataMessageHandler._
+import org.bitcoins.rpc.client.v19.BitcoindV19RpcClient
+import org.bitcoins.rpc.config.BitcoindInstance
+import org.bitcoins.testkit.BitcoinSTestAppConfig
+import org.bitcoins.testkit.wallet.BitcoinSWalletTest
+import org.bitcoins.wallet.Wallet
+import org.bitcoins.wallet.config.WalletAppConfig
+
+import scala.concurrent.{ExecutionContextExecutor, Future}
+
+implicit val system: ActorSystem = ActorSystem(s"node-api-example")
+implicit val ec: ExecutionContextExecutor = system.dispatcher
+implicit val walletConf: WalletAppConfig =
+    BitcoinSTestAppConfig.getSpvTestConfig().walletConf
+
+// let's use a helper method to get a v19 bitcoind
+// and a ChainApi
+  val bitcoind = BitcoindV19RpcClient(BitcoindInstance.fromConfigFile())
+val chainApi = BitcoinSWalletTest.MockChainQueryApi
+
+// Create our key manager
+  val keyManagerE = BIP39KeyManager.initialize(kmParams = walletConf.kmParams,
+                                               bip39PasswordOpt = None)
+
+val keyManager = keyManagerE match {
+    case Right(keyManager) => keyManager
+    case Left(err) =>
+      throw new RuntimeException(s"Cannot initialize key manager err=$err")
+  }
+
+// This function can be used to create a callback for when our node api calls downloadBlocks,
+// more specifically it will call the function every time we receive a block, the returned
+// NodeCallbacks will contain the necessary items to initialize the callbacks
+  def createCallback(processBlock: Block => Unit): NodeCallbacks = {
+    lazy val onBlock: OnBlockReceived = { block =>
+      processBlock(block)
+    }
+    NodeCallbacks(onBlockReceived = Seq(onBlock))
+  }
+
+// Here is a super simple example of a callback, this could be replaced with anything, from
+// relaying the block on the network, finding relevant wallet transactions, verifying the block,
+// or writing it to disk
+  val exampleProcessBlock = (block: Block) =>
+    println(s"Received block: ${block.blockHeader.hashBE}")
+val exampleCallback = createCallback(exampleProcessBlock)
+
+// Here is where we are defining our actual node api, Ideally this could be it's own class
+// but for the examples sake we will keep it small.
+  val nodeApi = new NodeApi {
+    override def downloadBlocks(
+        blockHashes: Vector[DoubleSha256Digest]): Future[Unit] = {
+      val blockFs = blockHashes.map(hash => bitcoind.getBlockRaw(hash))
+      Future.sequence(blockFs).map(_ => ())
+    }
+  }
+
+// Finally, we can initialize our wallet with our own node api
+  val wallet =
+    Wallet(keyManager = keyManager, nodeApi = nodeApi, chainQueryApi = chainApi)
```

Comment:
can you trigger this event so that `println(s"Received block: ${block.blockHeader.hashBE}")` is executed.

that should wrap up the example IMO.